java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java

```diff
@@ -625,7 +626,7 @@ public State scan(Tree tree, Void p) {
             
             TypeMirror currentType = tree != null ? info.getTrees().getTypeMirror(new TreePath(getCurrentPath(), tree)) : null;
             
-            if (currentType != null && currentType.getKind().isPrimitive()) {
+            if ((tree != null && tree instanceof JCTree.JCLambda) || (currentType != null && currentType.getKind().isPrimitive())) {
```

Comment:
You can use tree.getKind in place of instanceof

Reply:
in fact `getKind` is strongly preferred, as it allows the implementor to mix in interfaces. Please void dependencies on `JC*` implementation classes from Javac, if possible.

Reply:
getKind() method added.